src/frontend/src/pages/TeamsPage/DeleteTeamModalContainer/DeleteTeamView.tsx

```diff
@@ -0,0 +1,125 @@
+import { yupResolver } from '@hookform/resolvers/yup';
+import { useForm } from 'react-hook-form';
+import { useSingleTeam } from '../../../hooks/teams.hooks';
+import LoadingIndicator from '../../../components/LoadingIndicator';
+import ErrorPage from '../../ErrorPage';
+import * as yup from 'yup';
+import { DeleteTeamInputs } from './DeleteTeam';
+import {
+  Dialog,
+  DialogActions,
+  DialogContent,
+  DialogTitle,
+  FormControl,
+  FormLabel,
+  IconButton,
+  Typography
+} from '@mui/material';
+import CloseIcon from '@mui/icons-material/Close';
+import NERFailButton from '../../../components/NERFailButton';
+import NERSuccessButton from '../../../components/NERSuccessButton';
+import ReactHookTextField from '../../../components/ReactHookTextField';
+
+interface DeleteTeamViewProps {
+  teamId: string;
+  showModal: boolean;
+  onHide: () => void;
+  onSubmit: (data: DeleteTeamInputs) => Promise<void>;
+}
+
+const DeleteTeamView: React.FC<DeleteTeamViewProps> = ({ teamId, showModal, onHide, onSubmit }: DeleteTeamViewProps) => {
+  const { isLoading, isError, data: team, error } = useSingleTeam(teamId);
+
+  const teamNameTester = (teamName: string | undefined) =>
+    team !== undefined && teamName !== undefined && teamName.toLowerCase() === team.teamName.toLowerCase();
+
+  const schema = yup.object().shape({
+    teamName: yup.string().required().test('team-name-test', 'Team name does not match', teamNameTester)
+  });
+
+  const {
+    handleSubmit,
+    control,
+    formState: { errors, isValid }
+  } = useForm({
+    resolver: yupResolver(schema),
+    defaultValues: {
+      teamName: ''
+    },
+    mode: 'onChange'
+  });
+
+  if (isLoading || !team) return <LoadingIndicator />;
+  if (isError) return <ErrorPage message={error.message} />;
+
+  const onSubmitWrapper = async (data: DeleteTeamInputs) => {
+    await onSubmit(data);
+  };
+
+  return (
+    <Dialog open={showModal} onClose={onHide}>
```

Comment:
we have a component for modals now: NERFormModal, if you use that, you should be able to put all the code in just one file: DeleteTeamModal.tsx